### PR TITLE
Cyborg health alert and damage examining

### PR DESCRIPTION
**Pull request by Doru991**
## About the PR
This PR makes it so that cyborgs have their own health indicator different from organics, and can be examined in order to tell how damaged they are.
They also make a distinct sound when sufficiently damaged, to ~~annoy people~~ make sure everyone around knows they may be in distress.

## Why / Balance
It's completely impossible to tell if a cyborg is damaged without using a health analyzer, and I don't think that makes much sense. Other players should be able to tell roughly what shape a cyborg is in by looking at it, like they would with any other mob.
The cyborg itself, of course, should be able to tell how damaged it is, so that it can more accurately perform risk assessment and know when to request repairs.

## Technical details

Added `HealthAlert`, `CritAlert`, `DeadAlert` fields to `MobThresholdComponent` in order to be able to specify custom health/crit/dead HUD indicators for specific mobs, with the human alert as the default. Also added `ShowOverlays` field for when it wouldn't thematically make sense to have them.
Modified `MobThresholdSystem` to use the alerts specified inside the component rather than always use the human versions when showing alerts on-screen.
Modified `DamageOverlayUiController` to be able to hide the overlays when requested to via `MobThresholdComponent`.
## Media

https://github.com/space-wizards/space-station-14/assets/75124791/2d8eeac6-aca1-437b-99c7-565d11474ec1

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- tweak: It is now easier to tell how damaged cyborgs are.

### CODE REVIEW

File: Content.Client/UserInterface/Systems/DamageOverlays/DamageOverlayUiController.cs [hunk 1]
```diff
@@ -79,9 +79,17 @@ private void UpdateOverlays(EntityUid entity, MobStateComponent? mobState, Damag
...
         if (!_mobThresholdSystem.TryGetIncapThreshold(entity, out var foundThreshold, thresholds))
             return; //this entity cannot die or crit!!
+
+        if (!thresholds.ShowOverlays)
+        {
+            _overlay.BruteLevel = 0;
+            _overlay.CritLevel = 0;
+            _overlay.DeadLevel = 0;
+            return; //this entity intentionally has no overlays
+        }
```
Comment (metalgearsloth) [inline Content.Client/UserInterface/Systems/DamageOverlays/DamageOverlayUiController.cs:91]: Can't you just call ClearOverlay instead.

File: Content.Shared/Mobs/Components/MobThresholdsComponent.cs [hunk 2]
```diff
@@ -18,6 +19,25 @@ public sealed partial class MobThresholdsComponent : Component
...
+    /// </summary>
+    [DataField("healthAlert")]
+    public AlertType HealthAlert = AlertType.HumanHealth;
+
+    [DataField("critAlert")]
+    public AlertType CritAlert = AlertType.HumanCrit;
+
+    [DataField("deadAlert")]
+    public AlertType DeadAlert = AlertType.HumanDead;
+
```
Comment (metalgearsloth) [inline Content.Shared/Mobs/Components/MobThresholdsComponent.cs:34]: This should just be Dictionary<MobState, Alert> to be significantly more flexible.

File: Resources/Audio/Machines/attributions.yml [hunk 3]
```diff
@@ -37,3 +37,8 @@
   license: "CC0-1.0"
   copyright: "Modified from https://freesound.org/people/Pagey1969/sounds/566048/"
   source: "https://github.com/space-wizards/ss14-raw-assets/tree/99fcffb02d9953f031991fc8f9980f0c3abd3803/Audio"
+  
+- files: ["warning_buzzer.ogg"]
+  license: "CC-BY-NC-SA-3.0"
+  copyright: "Taken from TG station."
```
Comment (metalgearsloth) [inline Resources/Audio/Machines/attributions.yml:43]: TG isn't NC

Comment (deltanedas) [reply Resources/Audio/Machines/attributions.yml]: also the commit...

Comment (Doru991) [reply Resources/Audio/Machines/attributions.yml]: ok, I changed this one to something off of freesound but I'll keep that in mind

Comment (EmoGarbage404) [reply Resources/Audio/Machines/attributions.yml]: you didn't need to change the sound. The only thing you had to change was the license to `CC-BY-SA-3.0`.

File: Resources/Prototypes/Entities/Mobs/Cyborgs/base_borg_chassis.yml [hunk 4]
```diff
@@ -124,6 +136,15 @@
...
     thresholds:
+    - trigger:
+        !type:DamageTrigger
+        damage: 50
+      behaviors:
+      - !type:PlaySoundBehavior
+        sound:
+          path: /Audio/Machines/warning_beep.ogg
+          params:
+            volume: 10
```
Comment (EmoGarbage404) [inline Resources/Prototypes/Entities/Mobs/Cyborgs/base_borg_chassis.yml:147]: does this play only when taking damage? it might play on repair.

Comment (Doru991) [reply Resources/Prototypes/Entities/Mobs/Cyborgs/base_borg_chassis.yml:147]: I tested in-game by damaging then repairing a borg and damaging it again and it played both times the borg's damage went above 50, and didn't when it was healed.

Comment (Doru991) [reply Resources/Prototypes/Entities/Mobs/Cyborgs/base_borg_chassis.yml:147]: I'm pretty sure repairing a borg fully heals it at least currently, if it somehow was repaired and still had over 50 damage it possibly might play it but that's not something that currently happens

File: Resources/Audio/Machines/attributions.yml (hunk 3 above)
Comment (Doru991) [reply Resources/Audio/Machines/attributions.yml]: Oh I didn't even notice that detail

**Pull request merged**